### PR TITLE
resource types spli + commands in progress

### PR DESCRIPTION
**Pull request by F0urth**
And generally please use file scope namespaces

### CODE REVIEW

File: GrindingCity.Core/Extensions/Enum/MapEnumName.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿using GrindingCity.Domain.Entities.Resource.Enums;
+using GrindingCity.Domain.Entities.Resources.Enums;
+
+namespace GrindingCity.Core.Extensions.Enum
+{
+    public static class MapEnumName
+    {
+        private static readonly Dictionary<string, RawResourcesNames> RawResources = new Dictionary<string, RawResourcesNames>
```
Comment (F0urth) [inline GrindingCity.Core/Extensions/Enum/MapEnumName.cs:8]:
```suggestion
        private static readonly Dictionary<string, RawResourcesNames> RawResources = new()
```

File: GrindingCity.Core/Extensions/Enum/MapEnumName.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            { "Wheat", RawResourcesNames.Wheat },
+            { "Wood", RawResourcesNames.Wood }
+        };
+
+        private static readonly Dictionary<string, EndResourceNames> EndResources = new Dictionary<string, EndResourceNames>
```
Comment (F0urth) [inline GrindingCity.Core/Extensions/Enum/MapEnumName.cs:14]:
```suggestion
        private static readonly Dictionary<string, EndResourceNames> EndResources = new()
```

File: GrindingCity.Core/Extensions/Enum/MapEnumName.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            { "Beer", EndResourceNames.Beer }
+        };
+
+        public static RawResourcesNames MapRawEnumName(string name)
+        {
+            RawResources.TryGetValue(name, out RawResourcesNames result);
```
Comment (F0urth) [inline GrindingCity.Core/Extensions/Enum/MapEnumName.cs:21]: We can use in this case `Enum.TryParse<RawResourcesNames>(name, out var result);`

Comment (F0urth) [reply GrindingCity.Core/Extensions/Enum/MapEnumName.cs:21]: No guard for not finding resource

File: GrindingCity.Core/Extensions/Enum/MapEnumName.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+            return result;
+        }
+
+        public static EndResourceNames MapEndEnumName(string name)
```
Comment (F0urth) [inline GrindingCity.Core/Extensions/Enum/MapEnumName.cs:26]: Same as for `RawResourcesNames`

Comment (F0urth) [reply GrindingCity.Core/Extensions/Enum/MapEnumName.cs:26]: We could think about passing Enum from Api if we don't have good argument against it

File: GrindingCity.Core/Resource/Commands/Create/CreateResourceCommand.cs [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
 namespace GrindingCity.Core.Resource.Commands.Create
 {
-    public sealed class CreateResourceCommand : IRequest<IResource>
+    public sealed class CreateResourceCommand : IRequest
     {
         public Guid BuildingId { get; set; }
+        public ResourceType ResourceType { get; set; }
         public string Name { get; set; }
-        public int Amount { get; set; }
+        public int Amount { get; set; } = 0;
```
Comment (F0urth) [inline GrindingCity.Core/Resource/Commands/Create/CreateResourceCommand.cs:11]: `= 0;` Not needed

File: GrindingCity.Core/Resource/Commands/Create/CreateResourceCommand.cs [hunk 6]
```diff
@@ -1,12 +1,13 @@
-﻿using GrindingCity.Domain.Interfaces.Entities;
+﻿using GrindingCity.Domain.Entities.Resource.Enums;
 using MediatR;
 
 namespace GrindingCity.Core.Resource.Commands.Create
 {
-    public sealed class CreateResourceCommand : IRequest<IResource>
+    public sealed class CreateResourceCommand : IRequest
```
Comment (F0urth) [inline GrindingCity.Core/Resource/Commands/Create/CreateResourceCommand.cs:6]: I believe all props can be `init`

File: GrindingCity.Core/Resource/Commands/Create/CreateResourceCommand.cs [hunk 7]
```diff
@@ -1,12 +1,13 @@
...
     {
         public Guid BuildingId { get; set; }
+        public ResourceType ResourceType { get; set; }
         public string Name { get; set; }
```
Comment (F0urth) [inline GrindingCity.Core/Resource/Commands/Create/CreateResourceCommand.cs:10]: Should be null suppressed

File: GrindingCity.Core/Resource/Commands/Create/CreateResourceCommandHandler.cs [hunk 8]
```diff
@@ -13,9 +17,31 @@ public CreateResourceCommandHandler(IResourcesRepository resourcesRepository)
             _resourcesRepository = resourcesRepository;
         }
 
-        //public Task<IResource> Handle(CreateResourceCommand command, CancellationToken cancellationToken)
-        //{
-        //    var resource = 
-        //}
+        public async Task Handle(CreateResourceCommand command, CancellationToken cancellationToken)
+        {
+            IResource resource;
```
Comment (F0urth) [inline GrindingCity.Core/Resource/Commands/Create/CreateResourceCommandHandler.cs:22]:
```suggestion
           IResource resource = command.ResourceType switch
            {
                ResourceType.Raw => new RawResourceEntity(command.BuildingId, MapEnumName.MapRawEnumName(command.Name),
                    command.Amount),
                ResourceType.End => new EndResourceEntity(command.BuildingId, MapEnumName.MapEndEnumName(command.Name),
                    command.Amount),
                _ => throw new ArgumentOutOfRangeException(nameof(command.ResourceType),"Resources out of domain range")
            };

            await _resourcesRepository.AddAsync(resource);
```

Comment (F0urth) [reply GrindingCity.Core/Resource/Commands/Create/CreateResourceCommandHandler.cs:22]: I don't know how to add multiple line suggestion 😀 but i would change body to smth like this

File: GrindingCity.Core/Resource/Commands/Update/UpdateResourceCommand.cs [hunk 9]
```diff
@@ -0,0 +1,10 @@
+﻿using MediatR;
+
+namespace GrindingCity.Core.Resource.Commands.Update
+{
+    public class UpdateResourceCommand : IRequest<bool>
```
Comment (F0urth) [inline GrindingCity.Core/Resource/Commands/Update/UpdateResourceCommand.cs:5]: Can be `sealed` can be `record` if not then at least should be `init` only setters

File: GrindingCity.Core/Resource/Commands/Update/UpdateResourceCommandHandler.cs [hunk 10]
```diff
@@ -0,0 +1,24 @@
+﻿using GrindingCity.Domain.Interfaces.Repositories;
+using MediatR;
+
+namespace GrindingCity.Core.Resource.Commands.Update
+{
+    public class UpdateResourceCommandHandler : IRequestHandler<UpdateResourceCommand, bool>
```
Comment (F0urth) [inline GrindingCity.Core/Resource/Commands/Update/UpdateResourceCommandHandler.cs:6]: Believe unfinish

File: GrindingCity.Domain/Entities/Resource/EndResourceEntity.cs [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public sealed class EndResourceEntity : ResourceEntity
+    {
+        [NotMapped]
+        public EndResourceNames EnumName { get; set; }
+
+        public EndResourceEntity(Guid buildingId, EndResourceNames name)
+        {
+            BuildingId = buildingId;
+            base.Name = name.ToString();
```
Comment (F0urth) [inline GrindingCity.Domain/Entities/Resource/EndResourceEntity.cs:15]: Not needed `base`

File: GrindingCity.Domain/Entities/Resource/EndResourceEntity.cs [hunk 12]
```diff
@@ -0,0 +1,24 @@
+﻿using GrindingCity.Domain.Entities.Resource.Enums;
+using GrindingCity.Domain.Entities.Resources;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace GrindingCity.Domain.Entities.Resource
+{
+    public sealed class EndResourceEntity : ResourceEntity
+    {
+        [NotMapped]
+        public EndResourceNames EnumName { get; set; }
```
Comment (F0urth) [inline GrindingCity.Domain/Entities/Resource/EndResourceEntity.cs:10]: Just don't like this solution. Nothing ties me to make it miss matching with name prop

File: GrindingCity.Domain/Entities/Resource/RawResourceEntity.cs [hunk 13]
```diff
@@ -1,15 +1,23 @@
 ﻿using GrindingCity.Domain.Entities.Resources.Enums;
+using System.ComponentModel.DataAnnotations.Schema;
 
 namespace GrindingCity.Domain.Entities.Resources
 {
     public sealed class RawResourceEntity : ResourceEntity
     {
-        public RawResourcesNames Name { get; set; }
+        [NotMapped]
+        public RawResourcesNames EnumName { get; set; }
```
Comment (F0urth) [inline GrindingCity.Domain/Entities/Resource/RawResourceEntity.cs:9]: Just don't like this solution. Nothing ties me to make it miss matching with name prop

File: GrindingCity.Domain/Entities/Resource/RawResourceEntity.cs [hunk 14]
```diff
@@ -1,15 +1,23 @@
...
 
-        public RawResourceEntity(int amount, RawResourcesNames name)
+        public RawResourceEntity(Guid buildingId, RawResourcesNames name) 
+        { 
+            BuildingId = buildingId;
+            base.Name = name.ToString();
```
Comment (F0urth) [inline GrindingCity.Domain/Entities/Resource/RawResourceEntity.cs:14]: Not needed `base`

File: GrindingCity.WebApi/Controllers/ResourcesController.cs [hunk 15]
```diff
@@ -19,18 +18,19 @@ public ResourcesController(IMediator mediator)
         }
 
         [HttpPost]
-        public async Task<ActionResult<IResource>> Post([FromBody] AddResourceToBuildingDto dto)
+        public async Task<ActionResult> Post([FromBody] AddResourceToBuildingDto dto)
```
Comment (F0urth) [inline GrindingCity.WebApi/Controllers/ResourcesController.cs:21]: Sad `Post` noise 😣

File: GrindingCity.WebApi/Controllers/ResourcesController.cs [hunk 16]
```diff
@@ -1,5 +1,4 @@
-﻿using GrindingCity.Core.Building.Commands.Create;
-using GrindingCity.Core.Resource.Commands.Create;
+﻿using GrindingCity.Core.Resource.Commands.Create;
 using GrindingCity.Domain.Interfaces.Entities;
```
Comment (F0urth) [inline GrindingCity.WebApi/Controllers/ResourcesController.cs:2]: No Needed

File: GrindingCity.Core/Extensions/Enum/MapEnumName.cs (hunk 2 above)
Comment (darkswede) [reply GrindingCity.Core/Extensions/Enum/MapEnumName.cs:14]: isn't that 'personal preferences'?

File: GrindingCity.Core/Extensions/Enum/MapEnumName.cs (hunk 3 above)
Comment (darkswede) [reply GrindingCity.Core/Extensions/Enum/MapEnumName.cs:21]: resource wont exist without the building, validation will be during adding resource to the building. but good point, true :)

File: GrindingCity.Core/Extensions/Enum/MapEnumName.cs (hunk 4 above)
Comment (darkswede) [reply GrindingCity.Core/Extensions/Enum/MapEnumName.cs:26]: thats actually the plan :)

File: GrindingCity.Core/Resource/Commands/Create/CreateResourceCommand.cs (hunk 5 above)
Comment (darkswede) [reply GrindingCity.Core/Resource/Commands/Create/CreateResourceCommand.cs:11]: agree

File: GrindingCity.Core/Resource/Commands/Create/CreateResourceCommandHandler.cs (hunk 8 above)
Comment (darkswede) [reply GrindingCity.Core/Resource/Commands/Create/CreateResourceCommandHandler.cs:22]: nice :)

File: GrindingCity.Core/Resource/Commands/Update/UpdateResourceCommandHandler.cs (hunk 10 above)
Comment (darkswede) [reply GrindingCity.Core/Resource/Commands/Update/UpdateResourceCommandHandler.cs:6]: indeed

File: GrindingCity.Domain/Entities/Resource/EndResourceEntity.cs (hunk 12 above)
Comment (darkswede) [reply GrindingCity.Domain/Entities/Resource/EndResourceEntity.cs:10]: i wanted to use one column in table, that is where string name comes from

File: GrindingCity.WebApi/Controllers/ResourcesController.cs (hunk 15 above)
Comment (darkswede) [reply GrindingCity.WebApi/Controllers/ResourcesController.cs:21]: will fix :)

File: GrindingCity.Core/Extensions/Enum/MapEnumName.cs (hunk 2 above)
Comment (F0urth) [reply GrindingCity.Core/Extensions/Enum/MapEnumName.cs:14]: It technically is **but** why waste paper and introduce additional characters if there is no need?

**Pull request closed without merging**